### PR TITLE
add os_distro_name and os_distro_revision

### PR DESCRIPTION
**Pull request by dhens**
update os_version device posture terraform docs for new linux fields

### CODE REVIEW

**Review by jacobbednarz — CHANGES_REQUESTED** [review]
you'll need to check out https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/registry-documentation.md on how to have this documentation updated as `docs` is the automatically generated version of these files.

**Comment by dhens** [comment]
> you'll need to check out https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/registry-documentation.md on how to have this documentation updated as `docs` is the automatically generated version of these files.

I see, my bad. Thanks!

File: templates/resources/device_posture_rule.md [hunk 1]
```diff
@@ -74,6 +78,8 @@ The input structure depends on the device posture rule type.
 
 - `version` = (Required) The operating system semantic version.
 - `operator` = (Required) The version comparison operator in (>,>=,<,<=,==)
+- `os_distro_name` = (Optional) The operating system excluding version information.
+- `os_distro_revision` = (Optional) The operating system version excluding OS name information or release name.
```
Comment (jacobbednarz) [inline templates/resources/device_posture_rule.md:82]:
```suggestion
- `version` - (Required) The operating system semantic version.
- `operator` - (Required) The version comparison operator in (>,>=,<,<=,==)
- `os_distro_name` - (Optional) The operating system excluding version information.
- `os_distro_revision` - (Optional) The operating system version excluding OS name information or release name.
```

File: templates/resources/device_posture_rule.md [hunk 2]
```diff
@@ -19,10 +19,14 @@ resource "cloudflare_device_posture_rule" "corporate_devices_posture_rule" {
...
   match {
-    platform = "mac"
+    platform = "linux"
   }
   input {
     id = cloudflare_teams_list.corporate_devices.id
+    version            = 1.0.0
+    operator           = <
+    os_distro_name     = ubuntu
+    os_distro_revision = 1.0.0
```
Comment (jacobbednarz) [inline templates/resources/device_posture_rule.md:29]: these fields aren't present in the schema or public API docs; so we won't be able to add them at this time.

**Review by jacobbednarz — CHANGES_REQUESTED** [review]
you're referencing non-existing API fields and schema values so this is unable to be merged. i'd suggest first getting the fields documentated then added to cloudflare-go before attempting to add them here.

**Comment by jacobbednarz** [comment]
these fields are now present in the Go library so you can add them to the schema, CRUD operations and the docs.

**Comment by jacobbednarz** [comment]
i've pushed a few fixes here but the test suite is still failing with a service error. do you mind looking into this and getting the tests in the right shape?

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareDevicePostureRule_" -count 1 -parallel 1 -timeout 120m -parallel 1
?       github.com/cloudflare/terraform-provider-cloudflare     [no test files]
=== RUN   TestAccCloudflareDevicePostureRule_SerialNumber
--- PASS: TestAccCloudflareDevicePostureRule_SerialNumber (20.79s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersion
--- PASS: TestAccCloudflareDevicePostureRule_OsVersion (17.22s)
=== RUN   TestAccCloudflareDevicePostureRule_OsDistroName
    resource_cloudflare_device_posture_rule_test.go:103: Step 1/1 error: Error running apply: exit status 1

        Error: error creating Device Posture Rule for account "f037e56e89293a057740de681ac9abbe": access.api.error.invalid_device_posture_integration_type (12094)

          with cloudflare_device_posture_rule.phnddkhwtf,
          on terraform_plugin_test.tf line 2, in resource "cloudflare_device_posture_rule" "phnddkhwtf":
           2: resource "cloudflare_device_posture_rule" "phnddkhwtf" {

--- FAIL: TestAccCloudflareDevicePostureRule_OsDistroName (3.37s)
=== RUN   TestAccCloudflareDevicePostureRule_OsDistroRevision
    resource_cloudflare_device_posture_rule_test.go:140: Step 1/1 error: Error running apply: exit status 1

        Error: error creating Device Posture Rule for account "f037e56e89293a057740de681ac9abbe": access.api.error.invalid_device_posture_integration_type (12094)

          with cloudflare_device_posture_rule.xihdkeyskf,
          on terraform_plugin_test.tf line 2, in resource "cloudflare_device_posture_rule" "xihdkeyskf":
           2: resource "cloudflare_device_posture_rule" "xihdkeyskf" {

--- FAIL: TestAccCloudflareDevicePostureRule_OsDistroRevision (3.36s)
=== RUN   TestAccCloudflareDevicePostureRule_DomainJoined
--- PASS: TestAccCloudflareDevicePostureRule_DomainJoined (24.96s)
=== RUN   TestAccCloudflareDevicePostureRule_Firewall
--- PASS: TestAccCloudflareDevicePostureRule_Firewall (15.92s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption (23.11s)
FAIL
FAIL    github.com/cloudflare/terraform-provider-cloudflare/internal/provider   109.388s
?       github.com/cloudflare/terraform-provider-cloudflare/version     [no test files]
FAIL
make: *** [testacc] Error 1
```

**Comment by jacobbednarz** [comment]
depends on cloudflare/cloudflare-go#1087

**Comment by jacobbednarz** [comment]
with the upstream PR shimmed in, tests are passing

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareDevicePostureRule_" -count 1 -parallel 1 -timeout 120m -parallel 1
?       github.com/cloudflare/terraform-provider-cloudflare     [no test files]
=== RUN   TestAccCloudflareDevicePostureRule_SerialNumber
--- PASS: TestAccCloudflareDevicePostureRule_SerialNumber (16.55s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersion
--- PASS: TestAccCloudflareDevicePostureRule_OsVersion (15.39s)
=== RUN   TestAccCloudflareDevicePostureRule_LinuxOsDistro
--- PASS: TestAccCloudflareDevicePostureRule_LinuxOsDistro (12.75s)
=== RUN   TestAccCloudflareDevicePostureRule_DomainJoined
--- PASS: TestAccCloudflareDevicePostureRule_DomainJoined (15.87s)
=== RUN   TestAccCloudflareDevicePostureRule_Firewall
--- PASS: TestAccCloudflareDevicePostureRule_Firewall (17.78s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption (17.27s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/provider   96.001s
?       github.com/cloudflare/terraform-provider-cloudflare/version     [no test files]
```

**Pull request merged**